packages/client/test/end-to-end/binary-publish.test.ts

```diff
@@ -0,0 +1,86 @@
+import { Wallet } from 'ethers'
+import { fastWallet, fetchPrivateKeyWithGas } from '@streamr/test-utils'
+import { waitForCondition, areEqualBinaries } from '@streamr/utils'
+import { StreamrClient } from '../../src/StreamrClient'
+import { Stream } from '../../src/Stream'
+import { createTestStream, createTestClient } from '../test-utils/utils'
+import { StreamPermission } from '../../src/permission'
+
+describe('binary publish', () => {
+
+    const PAYLOAD = new Uint8Array([1, 2, 3])
+
+    let publisherPk: string
+    let subscriberWallet: Wallet
+
+    let publisher: StreamrClient
+    let subscriber: StreamrClient
+    let stream: Stream
+
+    const TIMEOUT = 15 * 1000
+
+    beforeAll(async () => {
+        subscriberWallet = fastWallet()
+        publisherPk = await fetchPrivateKeyWithGas()
+    }, 30 * 1000)
+
+    describe('encrypted', () => {
```

Comment:
This test doesn't really seem to care whether the contents are encrypted or unencrypted but rather whether publish / subscribe works for public and private streams. Therefore I would suggest we name these as follows.

```suggestion
    describe('private stream', () => {
```